vault/logical_system.go

```diff
@@ -2602,11 +2604,20 @@ func (b *SystemBackend) handleEnableAuth(ctx context.Context, req *logical.Reque
 		Version:               pluginVersion,
 	}
 
-	resp, err := b.Core.handleDeprecatedMountEntry(ctx, me, consts.PluginTypeCredential)
-	if err != nil {
-		return handleError(err)
+	var resp *logical.Response
+
+	// Detect and handle deprecated auth methods
+	status, ok := b.Core.builtinRegistry.DeprecationStatus(me.Type, consts.PluginTypeCredential)
+	if ok {
+		_, err = b.Core.handleDeprecatedMountEntry(ctx, me, consts.PluginTypeCredential)
+		if err != nil {
+			return handleError(err)
+		}
 	}
 
+	// Always update the mount entry's deprecation status when enabling new auth methods
+	me.DeprecationStatus = status.String()
```

Comment:
The function we're in only gets called when the mount is created for the first time. However, the deprecation status could change every time the mount is reloaded. You can check for all places that `RunningVersion` is updated, which I think should cover all places we might need to update this value as well.

Reply:
Sounds good! This seems like the right approach, but it will also require revamping the way `handleDeprecatedMountEntry` works. Updates in progress.